src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -390,17 +390,18 @@ private Task BuildTreeForSnapshotAsync(IDependenciesSnapshot snapshot)
             }
 
             var nowait = SubmitTreeUpdateAsync(
```

Comment:
SubmitTreeUpdateAsync pushes the change to the queue, and those requests will be processed in the future.  Most tree provider doesn't wait on that call.  (Unless you want to wait to make sure the tree has been updated, you don't need wait).  Those requests will be processed one at a time based on the order.